File: src/commands/TrayCommands.ts
import { invoke } from '@tauri-apps/api/core';

/**
 * System Tray Commands
 */
export class TrayCommands {
  /**
   * Minimize window to tray
   * @returns Minimize result message
   */
  static async minimize(): Promise<string> {
    return invoke('minimize_to_tray');
  }

  /**
   * Restore window from tray
   * @returns Restore result message
   */
  static async restore(): Promise<string> {
    return invoke('restore_from_tray');
  }

  
  /**
   * Update tray menu
   * @param accounts Account email list
   * @returns Update result message
   */
  static async updateMenu(accounts: string[]): Promise<string> {
    return invoke('update_tray_menu_command', { accounts });
  }
}

File: src-tauri/src/commands/tray_commands.rs
use crate::system_tray::{update_tray_menu, SystemTrayManager};
use tauri::Manager;

/// Update tray menu (new command, for frontend to call)
#[tauri::command]
pub async fn update_tray_menu_command(
    app: tauri::AppHandle,
    accounts: Vec<String>,
) -> Result<String, String> {
    update_tray_menu(&app, accounts)?;
    Ok("Tray menu updated".to_string())
}

/// Minimize to tray
#[tauri::command]
pub async fn minimize_to_tray(app: tauri::AppHandle) -> Result<String, String> {
    let system_tray = app.state::<SystemTrayManager>();
    system_tray.minimize_to_tray(&app)?;
    Ok("Windows minimized to tray".to_string())
}

/// Restore from tray
#[tauri::command]
pub async fn restore_from_tray(app: tauri::AppHandle) -> Result<String, String> {
    let system_tray = app.state::<SystemTrayManager>();
    system_tray.restore_from_tray(&app)?;
    Ok("Windows restored".to_string())
}
